src/Fixer/Import/NoUnusedImportsFixer.php

```diff
@@ -233,9 +153,9 @@ private function removeUnusedUseDeclarations(Tokens $tokens, array $useDeclarati
         }
     }
 
-    private function removeUseDeclaration(Tokens $tokens, array $useDeclaration)
+    private function removeUseDeclaration(Tokens $tokens, NamespaceUseAnalysis $useDeclaration)
     {
-        for ($index = $useDeclaration['end'] - 1; $index >= $useDeclaration['start']; --$index) {
+        for ($index = $useDeclaration->getEndIndex() - 1; $index >= $useDeclaration->getStartIndex(); --$index) {
```

Comment:
I'm bit confused by the `NamespaceUseAnalysis` name. Analysis is kind of report on many cases with numbers, like how many use imports in the code, what is their order etc.

But here it looks like 1 group of tokens. Using `NamespaceUseTokenGroup` would be more clear

Reply:
That's a matter of perception. I like NamespaceUseAnalysis better: it's an analysis on a use statement. Just like a ArgumentAnalysis is an analysis of one argument. I can name it anything you want, but in the end : it is what it is :)